realm/realm-library/src/objectServer/java/io/realm/SyncConfiguration.java

```diff
@@ -92,6 +92,7 @@ private SyncConfiguration(File directory,
                                 RealmProxyMediator schemaMediator,
                                 RxObservableFactory rxFactory,
                                 Realm.Transaction initialDataTransaction,
+                                boolean readOnly,
```

Comment:
I wrote a builder based initialization and I find that having a N length constructor is pretty much unavoidable.